WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -231,76 +224,124 @@ private extension CollectOrderPaymentUseCase {
                                                                              onDismiss: onCompleted))
     }
 
+    func isOrderAwaitingPayment() -> Bool {
+        order.datePaid == nil
```

Comment:
You're right, I primarily used `datePaid` for consistency, and it would be great to use something more definitive. I am a bit concerned about using a timestamp with seconds as its smallest unit to detect changes, but I've confirmed that the web does change this timestamp whenever a change is made to the order. [It happens whenever an order is updated](https://github.com/woocommerce/woocommerce/blob/2fc02016327d7895af46601221056da3634c2bc5/plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php#L2472) using the CRUD functions for the database, including when metadata is changed.

Direct order entity updates won't automatically change the `dateModified`... but they are pretty long established now, and with High Performance Order Storage having been released, extensions have a big motivation to update old code to use the CRUD functions. I don't know of any reason why we would code around that.

I would like us to use `needsPayment` and similar, but unfortunately when we did it in the past, [it made a bunch of orders lose their `Collect Payment` button](https://github.com/woocommerce/woocommerce-ios/pull/7694). I've not dug in to it any more deeply since then, but given that it's working at the moment and consistent with Android, I don't feel like it's the best use of time to go in to that now.

> This links to the next question, Could we somehow have one source of truth -to be used everywhere- to know if the order is awaiting payment? it could be an extension in Order or a computed property. That way if we want to refactor it in the future changing the references will be easier.

Yes, that would be nice. I think right now, it would just be `datePaid == nil` though, and in future it will change to `needsPayment == true`, which is no easier or harder to refactor than with the existing uses of it. I see little need to expand these changes (about refreshing the order on payment retry) out to update the other places that check whether an order is paid already...

This is a good reminder about the use of `datePaid == nil`, but still not the right place to fix it, IMO. WDYT?